tests/select_for_update/tests.py

```diff
@@ -179,6 +179,20 @@ def test_for_update_requires_transaction_only_in_execution(self):
         with self.assertRaises(transaction.TransactionManagementError):
             list(people)
 
+    @skipUnlessDBFeature('supports_select_for_update_with_limit')
+    def test_select_for_update_with_limit(self):
+        other = Person.objects.create(name='Other')
+        with transaction.atomic():
+            person = Person.objects.all().order_by('pk').select_for_update()[1:2].get()
+            self.assertEqual(person, other)
+
+    @skipIfDBFeature('supports_select_for_update_with_limit')
+    def test_unsupported_select_for_update_with_limit(self):
+        msg = 'LIMIT/OFFSET not allowed with select_for_update on this database backend.'
```

Comment:
Assuming you use `NotSupportedError`, I think checking for the exception class is enough and is more robust.